sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/PortEvaluationStage.java

```diff
@@ -262,10 +268,25 @@ protected void setProtos(PodInfoBuilder podInfoBuilder, Protos.Resource resource
             .get(Constants.PORTS_RESOURCE_TYPE);
     Optional<Integer> dynamicPort = Optional.empty();
     if (availablePorts != null) {
-      dynamicPort = availablePorts.getRanges().getRangeList().stream()
-          .flatMap(r -> IntStream.rangeClosed((int) r.getBegin(), (int) r.getEnd()).boxed())
-          .filter(p -> !consumedPorts.contains(p))
-          .findFirst();
+      if (!spec.getRanges().isEmpty()) {
+        List<Integer> constrainedPorts = new ArrayList<>();
+        for (RangeSpec range : spec.getRanges()) {
+          constrainedPorts.addAll(
+              IntStream.range(range.getBegin(), range.getEnd())
```

Comment:
Looking at the reply to this comment from @takirala 
> 2000 and 5050 are allowed as well, leaving begin or end empty defines an open interval. also validated in validPortRangesTest()

Shouldn't we have a `IntStream.rangeClosed` which is inclusive of the `range.getEnd()` whereas `IntStream.range` is exclusive of `range.getEnd()`

I agree with @takirala's comment about making the docs clear about inclusive vs exclusive ranges.

Reply:
oh ya good catch i should change this